### ISSUE TITLE
Foreign key violation error on Capsule delete

### ISSUE THREAD

**Issue by rplevka** [SatelliteQE/robottelo#4117]
Creating this issue to mark down the notes about findings during investigation.

tl;dr - proxy removal seems to collide with another test `ContentViewTestCase/test_positive_update_version_multiple`

The following test has failed with no precedence:
tests.foreman.cli.test_capsule.CapsuleTestCase.test_positive_refresh_features_by_name
The error occurred during the capsule removal and it indicates that something managed to add the indirect reference to the proxy into LCE-related table.

```
2016-12-21 05:49:00 [app] [W] Action failed
 | PG::Error: ERROR:  update or delete on table "smart_proxies" violates foreign key constraint "katello_capsule_lifecycle_environments_capsule_fk" on table "katello_capsule_lifecycle_environments"
 | DETAIL:  Key (id)=(7) is still referenced from table "katello_capsule_lifecycle_environments".
```

Since this does not occur in isolated run, it must have been some concurrently running test.
The `foreman/production.log` shows 2 POST requests to the `organizations` endpoint just before the traceback as a clear sign of concurrent tests running at the same time.
```
2016-12-21 05:48:57 [app] [I] Started POST "/katello/api/organizations/846/environments" for ::1 at 2016-12-21 05:48:57 +0100
2016-12-21 05:48:57 [app] [I]   Rendered api/v2/taxonomies/show.json.rabl (231.7ms)
2016-12-21 05:48:57 [app] [I] Completed 200 OK in 263ms (Views: 218.8ms | ActiveRecord: 19.4ms)
2016-12-21 05:48:57 [app] [I] Processing by Katello::Api::V2::EnvironmentsController#create as JSON
2016-12-21 05:48:57 [app] [I]   Parameters: {"name"=>"s4HW6z", "prior_id"=>721, "api_version"=>"v2", "organization_id"=>"846", "environment"=>{"name"=>"s4HW6z", "prior_id"=>721}}
2016-12-21 05:48:58 [app] [I] Authorized user admin(Admin User)
2016-12-21 05:48:58 [app] [I]   Rendered /opt/theforeman/tfm/root/usr/share/gems/gems/katello-3.0.0.90/app/views/katello/api/v2/common/create.json.rabl within katello/api/v2/layouts/resource (19.0ms)
2016-12-21 05:48:58 [app] [I] Completed 200 OK in 99ms (Views: 17.7ms | ActiveRecord: 14.4ms)
2016-12-21 05:48:58 [app] [I] Started POST "/katello/api/v2/repositories" for 10.8.0.99 at 2016-12-21 05:48:58 +0100
...
```

The robottelo master log is a good place to look and one can clearly find out the test executing the POST requests by seearching by the org name.
Note, the robottelo console log won't find this piece, as it only logs stdout+stderr of the failed/error-ed tests.

**Comment by rplevka** [comment]
https://bugzilla.redhat.com/show_bug.cgi?id=1409573
